src/SIL.LCModel.Core/Phonology/PhonEnvRecognizer.cs

```diff
@@ -57,7 +59,7 @@ public void ResetSegments(string[] saSegments)
 		/// <param name="saNaturalClasses"></param>
 		public void ResetNaturalClasses(string[] saNaturalClasses)
 		{
-			m_parser.ResetNaturalClasses(saNaturalClasses);
+			m_parser!.ResetNaturalClasses(saNaturalClasses);
```

Comment:
I say here and the previous method we throw an exception indicating that the class wasn't initlized properly if `m_parser` is null, otherwise a useless null exception will be thrown instead.